kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h

```diff
@@ -251,7 +251,7 @@ class ResidualBasedBlockBuilderAndSolver
 
 
             //#pragma omp parallel for firstprivate(nconditions, LHS_Contribution, RHS_Contribution, EquationId ) schedule(dynamic, 1024)
-            #pragma omp parallel for  schedule(guided, 512)
+            #pragma omp for  schedule(guided, 512)

```

Comment:
sorry forgot to answer.
yes improvement is not negligible on 16 cores. 
[link](https://lh3.googleusercontent.com/-fReTPYRgRic/WMLI9dXzQ5I/AAAAAAAAdKU/jWLEiZXo85cPsfN0QJxg9l2wOEg_kuXNACL0B/h429/2017-03-10.png)